client/src/unifyfs-stdio.c

```diff
@@ -594,17 +599,20 @@ static int unifyfs_stream_read(
 
             /* read data from file into buffer */
             size_t bufcount;
-            int read_rc = unifyfs_fd_read(s->fd, current, s->buf, s->bufsize, &bufcount);
-            if (read_rc != UNIFYFS_SUCCESS) {
-                /* ERROR: read error, set error indicator and errno */
+            size_t read_rc = unifyfs_fd_read(s->fd, current, s->buf,
+                s->bufsize);
+            if (read_rc  == -1) {
```

Comment:
I think you mean for this check to be <0 rather than ==-1.  However, I don't think we want that either, because I don't think our UNIFYFS_ERROR_* values are negative.

Reply:
Oh, I see later you updated the return values from fd_read, as well.  Let me keep reviewing.